### ISSUE TITLE
explicit-outlives-requirements should remove entire `where` clause if it removes the only bound

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust#105150]
Given the following code: 

```rust
// run-pass
#![allow(dead_code)]
#![allow(unused_variables)]
#![feature(unsize, coerce_unsized)]

#[repr(packed)]
struct UnalignedPtr<'a, T: ?Sized>
    where T: 'a,
{
    data: &'a T,
}

fn main() {

    impl<'a, T, U> std::ops::CoerceUnsized<UnalignedPtr<'a, U>> for UnalignedPtr<'a, T>
        where
        T: std::marker::Unsize<U> + ?Sized,
        U: ?Sized,
    { }

    let arr = [1, 2, 3];
    let arr_unaligned: UnalignedPtr<[i32; 3]> = UnalignedPtr { data: &arr };
    let arr_unaligned: UnalignedPtr<[i32]> = arr_unaligned;
}
```

The current output is:

```

warning: outlives requirements can be inferred
 --> src/main.rs:8:11
  |
8 |     where T: 'a,
  |           ^^^^^ help: remove this bound
  |
  = note: requested on the command line with `-W explicit-outlives-requirements`
```

Rustfix will fail to apply this:

````
The following errors were reported:
error: expected `{` after struct name, found `,`
 --> src/main.rs:8:11
  |
8 |     where ,
  |           ^ expected `{` after struct name

error: aborting due to previous error
````

So we need to extend span to include the `where:` as well?

**Comment by nirmal-j-patel** [comment]
@rustbot release-assignment

Haven't been able to focus on this. Releasing assignment.

**Comment by reez12g** [comment]
Progress report: I'm working on it.
